src/pages/ReportSettingsPage.js

```diff
@@ -56,6 +56,7 @@ class ReportSettingsPage extends Component {
 
         this.validate = this.validate.bind(this);
         this.updatePolicyRoomName = this.updatePolicyRoomName.bind(this);
+        this.shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report) || ReportUtils.isArchivedRoom(this.props.report);
```

Comment:
I think we should pass this in as a prop rather than making a class variable here in the constructor.

Reply:
(FYI, I say this considering the 'Data flows down' rule we have) - check out our [React Style Guide](https://github.com/Expensify/Style-Guides/blob/main/react.md#rule-1-dataflowsdown) for more info!

Reply:
There are several places that navigate to ReportSettingsPage:

[log_nav_to_report_settings_page.txt](https://github.com/Expensify/App/files/10689191/log_nav_to_report_settings_page.txt)

I don't know in which case the room name requires to be editable. I feel like the logic for determining whether the room name should be editable depending on `reportId` passed belongs to `ReportSettingsPags`.

@yuwenmemon Please let me know your thoughts

Reply:
Oh hm yes because this is a page that comes directly from a route I see. Okay so nevermind me then! :ohnothing: